scalameta/trees/shared/src/main/scala/scala/meta/internal/trees/ParentChecks.scala

```diff
@@ -102,4 +102,8 @@ object ParentChecks {
   def TypeMethod(tree: Type.Method, parent: Tree, destination: String): Boolean = {
     parent.is[Type]
   }
+
+  def CaseEnum(tree: Defn, parent: Tree, destination: String) : Boolean= {
+    parent.is[Template] && (parent.parent.isEmpty || parent.parent.get.is[Defn.Enum])
```

Comment:
@samuelchassot If I remember correctly, the issue is that here `parent.parent` is always empty?

Reply:
Yes exactly 
@olafurpg I just try to print *parent.parent* and parse a little program with a *case* inside a *class*, it prints *None*

Reply:
That could make sense since this check is only run when setting the parent, at which point there is no `parent.parent`. Trees are constructed one node at a time and this check only runs when creating a new root node.

I'm not excited about updating checks for every `Stat` node, so I think it's fine to leave out this check instead.